Source/WebCore/platform/text/WritingMode.h

```diff
@@ -75,66 +75,56 @@ constexpr inline BlockFlowDirection writingModeToBlockFlowDirection(WritingMode
     return BlockFlowDirection::TopToBottom;
 }
 
-constexpr unsigned makeTextFlowInitalizer(BlockFlowDirection blockFlow, TextDirection direction)
-{
-    return static_cast<unsigned>(blockFlow) << 1 | static_cast<unsigned>(direction);
-}
-
 // Define the text flow in terms of the writing mode and the text direction. The first
 // part is the block flow direction and the second part is the inline base direction.
-enum TextFlow {
-    InlineEastBlockSouth = makeTextFlowInitalizer(BlockFlowDirection::TopToBottom, TextDirection::LTR),
-    InlineWestBlockSouth = makeTextFlowInitalizer(BlockFlowDirection::TopToBottom, TextDirection::RTL),
-    InlineEastBlockNorth = makeTextFlowInitalizer(BlockFlowDirection::BottomToTop, TextDirection::LTR),
-    InlineWestBlockNorth = makeTextFlowInitalizer(BlockFlowDirection::BottomToTop, TextDirection::RTL),
-    InlineSouthBlockEast = makeTextFlowInitalizer(BlockFlowDirection::LeftToRight, TextDirection::LTR),
-    InlineSouthBlockWest = makeTextFlowInitalizer(BlockFlowDirection::LeftToRight, TextDirection::RTL),
-    InlineNorthBlockEast = makeTextFlowInitalizer(BlockFlowDirection::RightToLeft, TextDirection::LTR),
-    InlineNorthBlockWest = makeTextFlowInitalizer(BlockFlowDirection::RightToLeft, TextDirection::RTL)
-};
+struct TextFlow {
+    BlockFlowDirection blockFlowDirection;
+    TextDirection inlineBaseDirection;
 
-constexpr inline TextFlow makeTextFlow(WritingMode writingMode, TextDirection direction)
-{
-    TextDirection inlineBaseDirection = direction;
-    if (writingMode == WritingMode::SidewaysLr)
-        inlineBaseDirection = direction == TextDirection::RTL ? TextDirection::LTR : TextDirection::RTL;
-    return static_cast<TextFlow>(makeTextFlowInitalizer(writingModeToBlockFlowDirection(writingMode), inlineBaseDirection));
-}
+    constexpr inline bool isReversed()
+    {
+        return inlineBaseDirection == TextDirection::RTL;
+    }
 
-constexpr unsigned TextFlowReversedMask = 1;
-constexpr unsigned TextFlowFlippedMask = 1 << 1;
-constexpr unsigned TextFlowVerticalMask = 1 << 2;
+    constexpr inline bool isFlipped()
+    {
+        return blockFlowDirection == BlockFlowDirection::BottomToTop
+            || blockFlowDirection == BlockFlowDirection::RightToLeft;
+    }
 
-constexpr inline bool isReversedTextFlow(TextFlow textflow)
-{
-    return textflow & TextFlowReversedMask;
-}
+    constexpr inline bool isVertical()
+    {
+        return blockFlowDirection == BlockFlowDirection::LeftToRight
+            || blockFlowDirection == BlockFlowDirection::RightToLeft;
+    }
```

Comment:
If you set up the BlockFlowDirection constants correctly, you should be able to use a single bit check here, too. That's more efficient than two conditional checks.

Reply:
I'd privilege explicitness over shortness here, I don't think a bit mask is necessarily faster.

Reply:
Bit masking is faster than conditional checks with potentially failed branch prediction.

Reply:
I guess you could cast to an int then bit mask , but I really prefer explicit checks here